sycl/source/detail/kernel_impl.cpp

```diff
@@ -93,6 +93,35 @@ bool kernel_impl::isCreatedFromSource() const {
   return MCreatedFromSource;
 }
 
+void kernel_impl::checkIfValidForNumArgsInfoQuery() const {
+  bool isBuiltIn = false;
+  std::string KernelName = get_info<info::kernel::function_name>();
+  if (KernelName != "") {
+    auto Devices = MKernelBundleImpl->get_devices();
+    for (const auto &Device : Devices) {
+      auto BuiltInKernles =
+          Device.get_info<info::device::built_in_kernel_ids>();
+      if (!BuiltInKernles.empty()) {
+        auto It = std::find_if(BuiltInKernles.begin(), BuiltInKernles.end(),
```

Comment:
[`std::any_of`](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of) can be used here instead:

```c++

if (any_of(...) {
  isBuiltin = true;
  break;
}
```

Reply:
0b4f909